apis/r/R/utils-arrow.R

```diff
@@ -384,21 +394,59 @@ get_domain_and_extent_dataframe <- function(tbl_schema, ind_col_names,
           ind_max_dom <- arrow_type_range(ind_col_type) - c(0, ind_ext)
         }
 
-        ind_cur_dom <- ind_max_dom
+        requested_slot <- domain[[ind_col_name]]
+        if (is.null(requested_slot)) {
+          ind_cur_dom <- ind_max_dom
+        } else {
+          ind_cur_dom <- requested_slot
+        }
+        # Core supports no domain specification for variable-length dims, which
+        # includes string/binary dims.
         if (ind_col_type_name %in% c("string", "large_utf8", "utf8")) ind_ext <- NA
 
         # https://github.com/single-cell-data/TileDB-SOMA/issues/2407
         if (.new_shape_feature_flag_is_enabled()) {
             if (ind_col_type_name %in% c("string", "utf8", "large_utf8")) {
-                aa <- arrow::arrow_array(c("", "", "", "", ""), ind_col_type)
+                if (is.null(requested_slot)) {
+                  aa <- arrow::arrow_array(c("", "", "", "", ""), ind_col_type)
+                } else {
+                  aa <- arrow::arrow_array(c("", "", "", requested_slot[[1]], requested_slot[[2]]), ind_col_type)
+                }
```

Comment:
Make this if/else assignment a bit more R-tastic
```suggestion
                aa <- if (is.null(requested_slot)) {
                  arrow::arrow_array(c("", "", "", "", ""), ind_col_type)
                } else {
                  arrow::arrow_array(c("", "", "", requested_slot[[1]], requested_slot[[2]]), ind_col_type)
                }
```